axelor-business-project/src/main/java/com/axelor/apps/businessproject/web/InvoicingProjectController.java

```diff
@@ -97,4 +52,61 @@ public void fillIn(ActionRequest request, ActionResponse response) throws Axelor
     invoicingProjectService.setLines(invoicingProject, project, 0);
     response.setValues(invoicingProject);
   }
+
+  @SuppressWarnings("unchecked")
+  public void generateInvoice(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+
+    InvoicingProject invoicingProject;
+    Invoice invoice;
+    if (request.getContext().get("_ids") != null) {
+      List<Long> invoiceIdList = new ArrayList<Long>();
+      List<Integer> ids = (List<Integer>) request.getContext().get("_ids");
+      for (int id : ids) {
+        invoicingProject = Beans.get(InvoicingProjectRepository.class).find(Long.valueOf(id));
```

Comment:
instead of finding by ids and excluding on search better just use filter instead of find